### PR TITLE
Brand new examples for chapter 5

### PR DESCRIPTION
**Pull request by andreaswittig**
Three brand new examples for chapter 5:

Deploying Etherpad on EC2 with ...

... CloudFormation, ASG, and User Data
... CodeDeploy
... Packer

### CODE REVIEW

File: chapter05/cloudformation.yaml [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    Properties:
+      TargetGroupARNs:
+      - !Ref LoadBalancerTargetGroup
+      LaunchTemplate:
+        LaunchTemplateId: !Ref LaunchTemplate
+        Version: !GetAtt 'LaunchTemplate.LatestVersionNumber'
+      MinSize: '1'
+      MaxSize: '1'
+      HealthCheckGracePeriod: 300
+      HealthCheckType: EC2 # TODO Change to ELB
```
Comment (michaelwittig) [inline chapter05/cloudformation.yaml:257]: TODO :)

File: chapter05/codedeploy.yaml [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+    Properties:
+      TargetGroupARNs:
+      - !Ref LoadBalancerTargetGroup
+      LaunchTemplate:
+        LaunchTemplateId: !Ref LaunchTemplate
+        Version: !GetAtt 'LaunchTemplate.LatestVersionNumber'
+      MinSize: 1
+      MaxSize: 1
+      HealthCheckGracePeriod: 300
+      HealthCheckType: EC2 # TODO Change to ELB
```
Comment (michaelwittig) [inline chapter05/codedeploy.yaml:255]: TODO

File: chapter05/etherpad.pkr.hcl [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  ]
+  
+  provisioner "shell" {
+    inline = [
+      "curl -fsSL https://rpm.nodesource.com/setup_14.x | sudo bash -",
+      "sudo yum install -y nodejs git",
+      "sudo mkdir /opt/etherpad-lite",
+      "sudo chown -R ec2-user:ec2-user /opt/etherpad-lite",
+      "cd /opt",
+      "git clone --depth 1 --branch 1.8.18 https://github.com/ether/etherpad-lite.git",
```
Comment (michaelwittig) [inline chapter05/etherpad.pkr.hcl:66]: please clone repo to AWSinAction org

File: chapter05/cloudformation.yaml [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+            trap '/opt/aws/bin/cfn-signal -e 1 --stack ${AWS::StackName} --resource AutoScalingGroup --region ${AWS::Region}' ERR
+
+            # Install nodejs and git
+            curl -fsSL https://rpm.nodesource.com/setup_14.x | bash -
+            yum install -y nodejs git
+
+            # Fetch, configure, and start Etherpad as non-root user
+            su ec2-user -c '
+            cd /home/ec2-user/
+            git clone --depth 1 --branch ${EtherpadVersion} https://github.com/ether/etherpad-lite.git
```
Comment (michaelwittig) [inline chapter05/cloudformation.yaml:239]: please clone repo to AWSinAction org

File: chapter05/packer.yaml [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+    Properties:
+      TargetGroupARNs:
+      - !Ref LoadBalancerTargetGroup
+      LaunchTemplate:
+        LaunchTemplateId: !Ref LaunchTemplate
+        Version: !GetAtt 'LaunchTemplate.LatestVersionNumber'
+      MinSize: 1
+      MaxSize: 1
+      HealthCheckGracePeriod: 300
+      HealthCheckType: EC2 # TODO Change to ELB
```
Comment (michaelwittig) [inline chapter05/packer.yaml:206]: TODO

**Review by michaelwittig — CHANGES_REQUESTED** [review]

File: chapter05/cloudformation.yaml (hunk 4 above)
Comment (andreaswittig) [reply chapter05/cloudformation.yaml:239]: Good point, thanks!

**Pull request merged**